docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+```terraform
+module "utils" {
+  source  = "pingidentity/utils/pingone"
+  version = "~> 0.1"
+
+  # ... other configuration parameters
+}
+```
+
+[Terraform Documentation Reference](https://developer.hashicorp.com/terraform/tutorials/configuration-language/versions)
+
+## Protect Service Configuration and Data
+
+### Protect Configuration and Data with the `lifecycle.prevent_destroy` Meta Argument
+
+While some resources are safe to remove and replace, there are some resources that, if removed, can result in data loss.
+
+It's recommended to use the `lifecycle.prevent_destroy` meta argument to protect against accidental destroy plans that might cause data to be lost.  You may also want to use the meta argument to prevent accidental removal of access policies and applications if dependent applications cannot be updated with Terraform in case of replacement.
+
+For example:
+```terraform
+resource "pingone_schema_attribute" "my_attribute" {
+  environment_id = pingone_environment.my_environment.id
+
+  name = "myAttribute"
+  
+  # ... other configuration parameters
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+```
+
+### Don't Commit Secrets to Source Control - Use Terraform Variables and Secrets Management
+
+When writing Terraform HCL, it may be tempting to write values that are sensitive (such as OpenID Connect Client Secrets, TLS private key data, service passwords) directly into the code.  There is a significant risk that these secrets are then committed to source control, where they are able to be viewed by anyone who can access that code.  Even more so when the source control is a public Git repository hosted on sites such as Github or Gitlab.
+
+Committing secrets, such as passwords, API keys, and tokens, directly into Terraform configurations and subsequently into source control, poses significant security risks that can have far-reaching consequences. This practice exposes sensitive information to anyone who has access to the repository, including potential unauthorized users, thereby compromising the security of your infrastructure. Once secrets are committed to a repository, removing them requires extensive effort and does not guarantee that they haven't been copied or logged elsewhere.
+
+Moreover, version control systems are designed to track and preserve history, making it challenging to completely erase secrets once they are committed. This persistence in history means that even if the secrets are later removed from the codebase, they remain accessible in the commit history. Additionally, repositories, especially public or shared ones, are often cloned, forked, or integrated with third-party services, further increasing the exposure of secrets.
+
+Ultimately the safest way to recover from secrets that have been leaked is to rotate them in the source system, which can be an impactful activity if other systems or individuals depend on that credential.
```

Comment:
In the end, if credentials have been leaked in this manner, the safest way to recover is to rotate them in the source systems, an activity which can have broad impact across systems and individuals.